ci-operator/step-registry/ipi/conf/aws/windows-machineset/ipi-conf-aws-windows-machineset-commands.sh

```diff
@@ -10,7 +10,7 @@ oc wait deployment windows-machine-config-operator -n openshift-windows-machine-
 # Ensure userDataSecret exist, fail otherwise. The userDataSecret is required and contains specific information to
 # customize the machine at first boot. For instance, the authorized public key for the SSH server to accept
 # incoming connections, firewall rules, etc.
-oc -n openshift-machine-api get secret "${WINDOWS_USER_DATA_SECRET}"
+until oc -n openshift-machine-api get secret "${WINDOWS_USER_DATA_SECRET}" 2> /dev/null; do echo -n "." && sleep 2; done
```

Comment:
This is going to be an infinite loop if the secret was not created.
In that case this job will not fail until the 4ish hour timeout is hit.
There needs to be some reasonable limit to retries here, you can use a for loop for that.

Reply:
In addition please increase the time between API calls to 5 seconds.

Reply:
As suggested by @saifshaikh48 , I made use of the timeout function to ensure that if after 3 minutes the secret isn't created the step will fail.